### ISSUE TITLE
Advanced Engine Support

### ISSUE THREAD

**Issue by Javex** [Javex/libscep#55]
Once #26 is finished, engines can be used with our library more easily. Beyond this basic support, some issues may still arise in unusual cases.

For example, if an engine does not support RSA it is possible that our message functions will fail. But PKCS#7 allows to choose a different algorithm, so maybe it would be nice to check for this support and choose the used algorithms appropriately? It's also possible that OpenSSL already does that, we could check for that.

There might be more edge cases we wish to investigate. This issue exists to collect these cases and figure out how important certain features are.

Create a new issue for each of these features so we can keep this for just collecting.
